src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -239,10 +239,32 @@ public override async Task<CompletionChange> GetChangeAsync(Document document, C
                 assemblyName += $", PublicKey={ publicKey }";
             }
 
-            var textChange = new TextChange(item.Span, assemblyName);
+            var span = await GetTextChangeSpanAsync(document, item, cancellationToken);
+            var textChange = new TextChange(span, assemblyName);
             return CompletionChange.Create(textChange);
         }
 
+        private static async Task<TextSpan> GetTextChangeSpanAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var span = item.Span;
+            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(item.Span.Start);
+            if (syntaxFactsService.IsStringLiteral(token) && !string.IsNullOrEmpty(token.ValueText))
+            {
+                span = token.Span;
+                var positionOfValueIntText = token.Text.IndexOf(token.ValueText);
```

Comment:
nope.  do not do this.  it will not work in any sort of interesting cases (i.e. an escaped character).  In fact, please add a test that included an escaped character right now :)

Reply:
@MaStr11, please consider suggestions by @cyrus.najmabadi@gmail.com 

---
In reply to: [212066374](https://github.com/dotnet/roslyn/pull/29447#discussion_r212066374) [](ancestors = 212066374)